services/data_service.py

```diff
@@ -136,6 +137,39 @@ def batch_create_repo(self, repo: FirestoreRepo) -> str:
 
         return repo.id
 
+    def batch_delete_user_repo(self, user_id: str, repo_id: str) -> str:
+        repo = FirestoreRepo(**self.get_user_repo(user_id, repo_id))
+
+        if repo.owner != user_id:
```

Comment:
I think we already check this in `self.get_user_repo`

Reply:
yes good catch